corehq/ex-submodules/couchforms/management/commands/purge_forms_and_cases.py

```diff
@@ -0,0 +1,132 @@
+from dimagi.utils.django.management import are_you_sure
+
+
+X_FORM_HEADER = ['Version', 'Form Id', 'Domain']
+CASE_HEADER = ['CaseID', 'CaseType', 'CaseOpenedOn', 'CaseName', 'CaseUserId', 'CaseXFormIds']
+X_FORM_FILENAME = 'X-Form-Details.csv'
+CASE_FILE_NAME = 'Case-Details.csv'
+
+
+# Expects 4 arguments in order : domain app_id version_number test_run
+# ex: ./manage.py purge_forms_and_cases testapp c531daeece0633738c9a3676a13e3d4f 88 yes
+# domain is included with app_id to ensure the user knows what app to delete
+# version_number to delete data accumulated by versions BEFORE this : integer
+# test_run should be yes(case-sensitive) for a test_run and anything otherwise
+# though deletion would be re-confirmed so no panic
+class Command(BaseCommand):
+    def __init__(self):
+        super(Command, self).__init__()
+        self.case_ids = set()
+        self.filtered_xform_ids, self.xform_ids = [], []
+        self.xform_writer, self.case_writer = None, None
+        self.forms_accessor, self.case_accessors = None, None
+        self.domain, self.app_id, self.version_number, self.test_run = None, None, None, None
+
+    def setup(self):
+        self.xform_writer = csv.writer(open(X_FORM_FILENAME, 'w+b'))
+        self.xform_writer.writerow(X_FORM_HEADER)
+        self.case_writer = csv.writer(open(CASE_FILE_NAME, 'w+b'))
+        self.case_writer.writerow(CASE_HEADER)
+        self.forms_accessor = FormAccessors(self.domain)
+        self.case_accessors = CaseAccessors(self.domain)
+
+    def ensure_prerequisites(self, args):
+        if len(args) != 4:
+            raise CommandError('Arguments number mismatch')
+        self.domain = args[0]
+        self.app_id = args[1]
+        self.version_number = int(args[2])
+        self.test_run = True if args[3] == 'yes' else False
+        _notify_parsed_args(*args)
+        app = Application.get(self.app_id)
+        if app.domain != self.domain:
+            raise CommandError('Domain not same as from app id')
+        self.setup()
+
+    def handle(self, *args, **options):
+        self.ensure_prerequisites(args)
+        self.xform_ids = self.forms_accessor.get_all_form_ids_in_domain()
+        self.iterate_forms_and_collect_case_ids()
+        _print_final_debug_info(self.xform_ids, self.filtered_xform_ids, self.case_ids)
+        if self.data_to_delete() and self.delete_permitted():
+            self.delete_forms_and_cases()
+            print "Process Completed!! Keep copy of files %s, %s" % (X_FORM_FILENAME, CASE_FILE_NAME)
+        else:
+            print 'Process Finished w/o Changes..'
+
+    def iterate_forms_and_collect_case_ids(self):
+        print 'Iterating Through X-Forms and Collecting Case Ids'
+        for xform in self.forms_accessor.iter_forms(self.xform_ids):
+            _print_form_details(xform, self.xform_writer)
+            if '@version' in xform.form and int(xform.form['@version']) < self.version_number:
```

Comment:
because of the way form versioning works this isn't going to be the same as the app version. Have to use `Application.get(form.build_id).version` instead (and should probably keep a cache of this to avoid multiple lookups per app)